webrtc-stats.html

```diff
@@ -4249,6 +4240,25 @@ <h2>Obsolete RTCInboundRtpStreamStats members</h2>
             </dd>
           </dl>
         </section>
+        <pre class="idl">partial dictionary RTCMediaHandlerStats {
+          boolean             remoteSource;
+};</pre>
+        <section>
+          <h2>Obsolete RTCMediaHandlerStats members</h2>
+          <dl data-dfn-for="RTCMediaHandlerStats">
+            <dt>
+              <dfn><code>remoteSource</code></dfn> of type <span class=
+                "idlMemberType">boolean</span>
+            </dt>
+              <dd>
+                This field got removed in June 2019, as it was redundant and
+                sometimes ambiguous (when a track from one peer connection is
+                fed into another). A <code>type</code> of
```

Comment:
I don't see how it's ambiguous. A track fed from one PC to another PC would have remoteSource = true; remoteSource says nothing about its destination.

Reply:
remoteSource means the stats are "receiver" stats (as opposed to "sender" stats). This is needed for legacy implementations (like current Chrome) that have "track" stats in both sending and receiving directions. Can we add a receiver track attachment dictionary that is inheriting from the receiver stats dictionaries that only exists in the obsolete section? And have the only new member to this dictionary the remoteSource?

In this case "remoteSource" is not redundant, because the type is "track" for both sending and receiving directions. This is how Chrome works today, so we can't remove remoteSource.

As for "ambiguous (when a track from one peer connection fed into another)" I don't understand this. remoteSource means its either a sender track attachment or a receiver track attachment. The "track" stats object _does not_ refer to the track; only to the attachment. It matters not whether the track came from getUserMedia() or from another peer connection; the attachment is "remoteSource == false" if the attachment refers to an RTCRtpSender, and "remoteSource == true" if the attachment refers to an RTCRtpReceiver.

Reply:
Another way to think of remoteSource is "did the MediaStreamTrack that the trackIdentifier refers to get created by this RTCPeerConnection or by other means?"